hook-common/src/pgqueue.rs

```diff
@@ -527,13 +569,15 @@ impl PgQueue {
             pool,
             retry_policy,
             table,
-            worker,
         })
     }
 
     /// Dequeue a Job from this PgQueue to work on it.
-    pub async fn dequeue<J: DeserializeOwned + std::marker::Send + std::marker::Unpin + 'static>(
+    pub async fn dequeue<
```

Comment:
> I take it the idea is to try the txn-based one first and swap this by code changes in if we have issues?

Yes, although we could also make this configurable with, e.g., a `enable_transactional` key.

> It's odd to me that we close the connection if no rows are returned, that doesn't seem like an error? I think we could drop it so it (I assume) returns to the connection pool for reuse?

We can let it drop, I am following the docs and being explicit with `close`:

> This method is not required for safe and consistent operation. However, it is recommended to call it instead of letting a connection drop as the database backend will be faster at cleaning up resources.

I now **think** this was meant to describe cleaning-up a non-pooled connection. 

> I don't think we want to return the Connection in the PgJob for non-txn dequeue, there's no reason to tie up the PG connection for the lifetime of the webhook send, right?

True, we can pass the pool instead. Some lifetime wrangling may be required, but should be doable.

Reply:
`dequeue_tx` _could_ just `DELETE` rows when it's done -- but I don't know what we have in mind for customer facing metrics. If we're going to bill for this, I assume we need to show them success/fail as graphs or something? It seems convenient for a scheduled job to scrape the jobs table and push those into CH as it does deletes, or something. I'm just riffing, maybe we had other plans? (I was out when this project was discussed.)

btw, I imagine River is just trying to keep its dependencies minimal (i.e. make no expectations about k8s or anything). For our needs we could run a singleton k8s pod that has a separate `hook-cron` process that runs every minute or something, right?